core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -537,7 +537,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
 
     _executorAllocationManager =
-      if (dynamicAllocationEnabled) {
+      if (dynamicAllocationEnabled &&
+        (!isLocal || conf.getBoolean("spark.dynamicAllocation.testing", false))) {
```

Comment:
this is hard to read. Why not just make the change in `Utils.isDynamicAllocationEnabled` so all usages of it will see this change as well?